docs/rest-api.md

```diff
@@ -0,0 +1,909 @@
+# Flink SQL Gateway REST API
+
+## Error Handling
+
+When the REST API encounters an error, it will return a JSON object containing only one field "errors", a string array in which the strings are the details of the errors. This behavior reuses the error handling in the original Flink REST server code. We will not apply additional "error" fields in the following APIs.
+
+Currently, the HTTP response code when encountering an error will be either 400 BAD REQUEST (if the parameters provided by the user is invalid) or 500 INTERNAL SERVER ERROR (otherwise).
+
+## Basic JSON Objects
+
+### Result Set
+
+A result set is a JSON object indicating the execution results of a statement or a job.
+
+<table>
+  <tr>
+    <th>Field</th>
+    <th>Type</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>result_kind</td>
+    <td>string</td>
+    <td><p>Type of the result set, which is either <code>SUCCESS</code> or <code>SUCCESS_WITH_CONTENT</code>.</p><ul><li><code>SUCCESS</code> is just an acknowledgement indicating that your request is successfully handled.</li><li><code>SUCCESS_WITH_CONTENT</code> indicates that there are more information in the result set.</li></ul></td>
+  </tr>
+  <tr>
+    <td>columns</td>
+    <td>array of column type</td>
+    <td>The i-th element in the array indicates the type of the i-th column of this result set.</td>
+  </tr>
+  <tr>
+    <td>data</td>
+    <td>array of array</td>
+    <td>The i-th element (which is also an array) in the array indicates the i-th result row of this result set.</td>
+  </tr>
+  <tr>
+    <td>change_flags</td>
+    <td>array of boolean</td>
+    <td><p>This field only appears for a streaming job result.</p><ul><li>If the i-th element is `true`, the i-th result row should be appended to the results of the job.</li><li>If the i-th element is `false`, the i-th result row should be retracted from the results of the job.</li></ul></td>
```

Comment:
This field only appears for a streaming select job result.

Reply:
mark this as PublicEvolving, because flink-1.11 has updated this to [RowKind](https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/types/RowKind.java)